lib/topological_inventory/ansible_tower/operations/worker.rb

```diff
@@ -40,6 +49,14 @@ def queue_opts
           TopologicalInventory::AnsibleTower::MessagingClient.default.worker_listener_queue_opts
         end
 
+        def async_worker
+          @async_worker ||= TopologicalInventory::Providers::Common::Operations::AsyncWorker.new(async_queue, Processor).run
+        end
+
+        def async_queue
```

Comment:
The Queue can be part of AsyncWorker

Reply:
Oooh yeah I like that idea, changed the arity of the constructor to make the `queue` arg optional.